sdk/webpubsub/Microsoft.Azure.WebJobs.Extensions.WebPubSub/src/Trigger/WebPubSubTriggerBinding.cs

```diff
@@ -184,32 +187,32 @@ private object GetValueByName(string parameterName, Type targetType)
                     {
                         return value;
                     }
-                    return ConvertTypeIfPossible(value, targetType);
+                    return ConvertTypeIfPossible(value, targetType, _serializer);
                 }
                 return null;
             }
 
-            private static object ConvertTypeIfPossible(object source, Type target)
+            private static object ConvertTypeIfPossible(object source, Type target, JsonSerializer serializer)
             {
                 if (source is BinaryData data)
                 {
-                    return data.Convert(target);
+                    return data.Convert(target, serializer);
                 }
                 if (target == typeof(JObject))
                 {
-                    return JToken.FromObject(source);
+                    return JToken.FromObject(source, serializer);
                 }
                 if (target == typeof(string))
                 {
-                    return JToken.FromObject(source).ToString();
+                    return JToken.FromObject(source, serializer).ToString();
                 }
                 if (target == typeof(byte[]))
```

Comment:
Can the target be BinaryData?

Reply:
Yes. Included. Mainly works for the message data.

Reply:
I only see a check for BinaryData as a source.

Reply:
`data.Convert()` which calls to [`BinaryData.Convert`](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/webpubsub/Microsoft.Azure.WebJobs.Extensions.WebPubSub/src/Bindings/BinaryDataExtensions.cs#L33) helps to convert in need.

Reply:
Basically the target type should match its source type, which is handled before this `ConvertTypeIfPossible()`. And this method is to help convert to some basic types if user requires. `BinaryData` is not quite commonly used object type and doesn't see much needs to supports it for variety binding objects except for message binding object `Data`(source type is BinaryData), that is handled  in first `if` as source type is `BinaryData`.

Reply:
Maybe a more generic way is using string -> Json deserialize to support generic type.

Reply:
BinaryData is considered a simple type that customers may want to bind to in other extensions like EventGrid, Storage, Service Bus

Reply:
OK. I think we shouldn't encourage them do this cause it's not efficient for the extension to convert dynamic types to BinaryData. For example, to convert the `WebPubSubConnectionContext`, we have to JObject -> string -> BinaryData without a known `T` type to call `FromObjectAsJson<T>()`. And actually, user can handle this themselves more directly like below.

```cs
[FunctionName("message")]
public static async Task<UserEventResponse> Run(
    [WebPubSubTrigger(WebPubSubEventType.User, "message")] UserEventRequest request,
    WebPubSubConnectionContext connectionContext,
    BinaryData data,
    WebPubSubDataType dataType,
    [WebPubSub(Hub = "simplechat")] IAsyncCollector<WebPubSubAction> actions)
{
    var binaryContext = BinaryData.FromObjectAsJson<WebPubSubConnectionContext>(connectionContext);
    ...
}
```